pilot/pkg/xds/ads.go

```diff
@@ -480,6 +495,23 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *Connection) error
 	return nil
 }
 
+func authorizeConnection(con *Connection) error {
+	for _, rawID := range con.Identities {
+		spiffeID, err := spiffe.ParseIdentity(rawID)
+		if err != nil {
+			continue
+		}
+		if con.proxy.ConfigNamespace != "" && spiffeID.Namespace != con.proxy.ConfigNamespace {
+			continue
+		}
+		if con.proxy.Metadata.ServiceAccount != "" && spiffeID.ServiceAccount != con.proxy.Metadata.ServiceAccount {
```

Comment:
same here in what case ServiceAccount is empty?

Reply:
Probably on VMs and manual injection.

Same as above - this will remove another config for VM, and one less thing to populate in node metadata. 

I guess we can warn if it's not the same - but in any case just set it from identity.

Reply:
ServiceAccount is used only in one place today - https://github.com/istio/istio/blob/d91ce670963861378dbfa8134e214138f40159e9/pilot/pkg/serviceregistry/kube/controller/controller.go#L972 and I am almost certain we could remove it. However, in the future we may use it for VM autoregistration. It seems like the current approach is that we will look up the ServiceAccount from the WorkloadGroup though, so that likely will not be deleted. We can likely remove this part?